src/shogun/evaluation/MachineEvaluation.cpp

```diff
@@ -121,5 +165,19 @@ CMachine* CMachineEvaluation::get_machine() const
 
 EEvaluationDirection CMachineEvaluation::get_evaluation_direction()
 {
-	return m_evaluation_criterion->get_evaluation_direction();
+    REQUIRE(m_list_evaluation_criterion->get_array_size()==1,"Multiple Metrics provided. Please use get_evaluation_directions()");
```

Comment:
The REQUIRE is bad, please see https://github.com/shogun-toolbox/shogun/wiki/Assertions

It can segfault, and please do a proper error message that helps the user rather than confusing him